hybrid/power_source.py

```diff
@@ -36,6 +36,11 @@ def __init__(self, name, site: SiteInfo, system_model, financial_model):
         self._financial_model = financial_model
         self._layout = None
         self._dispatch = PowerSourceDispatch
+        try:
+            # Update system capacity in custom or user-instantiated financial models
+            self._financial_model.value("system_capacity", self._system_model.value("system_capacity"))
+        except:
+            pass
```

Comment:
I think there is the potential for there to be other variables that need to be copied over for custom financial models. Regardless, what needs to be copied over can be potentially extensive and specific to the model. I think we need to design this part of the code to make it explicit that this needs to happen, which is a change from the code now which kind of relies on how PySAM does it implicitly.

So I think there should be custom financial model function that's like "copy over all needed variables" and then in power_source, there could be a "check if this function exists, if so call it" and some comments about how PySAM does it automatically.

Reply:
I agree. This has been implemented in `set_financial_inputs` in 'custom_financial_model.py`, which is called in `PowerSource::__init__` and in the overloaded `execute` function for the custom financial model.

Reply:
Great